testsuite/src/test/java/org/ironjacamar/core/connectionmanager/pool/PrefillCriTestCase.java

```diff
@@ -0,0 +1,650 @@
+   private static UserTransaction ut;
+
+   /**
+    * The resource adapter
+    *
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 1)
+   private ResourceAdapterArchive createResourceAdapter() throws Throwable
+   {
+      return ResourceAdapterFactory.createUnifiedSecurityRar();
+   }
+
+
+
+   /**
+    * The activation W/o Tx
+    *
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 2)
+   private ResourceAdaptersDescriptor createNoTxActivation() throws Throwable
+   {
+      return ResourceAdapterFactory.createUnifiedSecurityDeployment(null, null,
+            TransactionSupport.TransactionSupportLevel.NoTransaction,
+            "UnifiedSecurityNoTxConnectionFactory", true, 10);
+   }
+
+   /**
+    * The activation W/ LocalTx
+    *
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 3)
+   private ResourceAdaptersDescriptor createLocalTxActivation() throws Throwable
+   {
+      return ResourceAdapterFactory.createUnifiedSecurityDeployment(null, null,
+            TransactionSupport.TransactionSupportLevel.LocalTransaction,
+            "UnifiedSecurityLocalTxConnectionFactory", true, 10);
+   }
+
+   /**
+    * The activation W/ XATx
+    *
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 4)
+   private ResourceAdaptersDescriptor createXATxActivation() throws Throwable
+   {
+      return ResourceAdapterFactory.createUnifiedSecurityDeployment(null, null,
+            TransactionSupport.TransactionSupportLevel.XATransaction,
+            "UnifiedSecurityXATxConnectionFactory", true, 10);
+   }
+
+
+   /**
+    * test w/o Tx, 2 calls w/ same credentials
+    *
+    * @throws Throwable In case of an error
+    */
+   @Test
+   public void testOneCredentialNoTxWithoutTransaction() throws Throwable
+   {
+      assertNotNull(noTxCf);
+      assertNotNull(dr);
+
+      assertEquals(3, dr.getDeployments().size());
+
+      org.ironjacamar.core.api.deploymentrepository.Deployment d = dr
+            .findByJndi("java:/eis/UnifiedSecurityNoTxConnectionFactory");
+      assertNotNull(d);
+
+      org.ironjacamar.core.api.deploymentrepository.ConnectionFactory dcf = d.getConnectionFactories().iterator()
+            .next();
+      assertNotNull(dcf);
+
+      org.ironjacamar.core.api.deploymentrepository.Pool p = dcf.getPool();
+      assertNotNull(p);
+
+      DefaultPool defaultPool = (DefaultPool) p.getPool();
+
+      ConcurrentHashMap<Credential, ManagedConnectionPool> mcps =
+            (ConcurrentHashMap<Credential, ManagedConnectionPool>) TestUtils
+                  .extract(defaultPool, "pools");
+      assertNotNull(mcps);
+      assertEquals(1, mcps.size());
+
+
+      UnifiedSecurityConnection c =  noTxCf.getConnection("user", "pwd");
+      assertNotNull(c);
+      assertEquals("user", c.getUserName());
+      assertEquals("pwd", c.getPassword());
+
+      UnifiedSecurityConnection c1 = noTxCf.getConnection("user", "pwd");
```

Comment:
We only need to test connections that are different than the prefill credentials